### ISSUE TITLE
Incorrect warning about undeclared variable inside trap

### ISSUE THREAD

**Issue by mikhailnov** [koalaman/shellcheck#2681]
Shellcheck incorrectly warns that "i is referenced but not assigned" (SC2154) for the following code:

```bash
#!/bin/bash
trap 'for i in $(tac "$PID_LIST"); do kill $i || : ; kill -9 $i || : ; done; rm -fr "$PID_LIST" "$CHROMIUM_PROFILES_DIR" "$SCROT_TMP" "$CDIR"' EXIT ERR
```

Of course it is a badly written code but the warning is incorrect.

The same code not inside trap is parsed correctly:
```bash
#!/bin/bash
for i in $(tac "$PID_LIST"); do kill $i || : ; kill -9 $i || : ; done; rm -fr "$PID_LIST" "$CHROMIUM_PROFILES_DIR" "$SCROT_TMP" "$CDIR"' 
```
SC2086 is correctly given.
